File: spec/TransactionSpec.js
describe('Transaction', function() {
    var transaction; 
    var client;
});

it('new instance should have balance of 0', function() {
    transaction = new Transaction();
    expect(transaction.balance).toEqual(0)
})

// beforeEach(function() {
//     transaction = new Transaction();
//     client = new Client();
//     client
// ;})

// it('should change credit and new balance in transaction', function() {
//     transaction.credit_account(50);
//     expect(transaction.credit).toEqual(50);
//     expect(transaction.debit).toEqual(0);
//     expect(transaction.balance).toEqual(50);
// });

// it('should change debit and balance using current balance', function() {
//     client.current_balance = 100
//     transaction.debit_account(client.current_balance, 25);
//     expect(transaction.credit).toEqual(0);
//     expect(transaction.debit).toEqual(25);
//     expect(transaction.balance).toEqual(75);
// });

File: spec/transaction_spec.rb
# frozen_string_literal: true

require 'transaction'

describe Transaction do
  it 'should display deposit transaction' do
    date = Time.new(2019, 2, 5)
    allow(Time).to receive(:now).and_return(date)
    transaction = Transaction.new(balance: 100, credit: 40)

    expect(transaction.display).to eq '05/02/2019 || 40.00 ||  || 100.00 '
  end

  it 'should display withdraw transaction' do
    date = Time.new(2019, 2, 5)
    allow(Time).to receive(:now).and_return(date)
    transaction = Transaction.new(balance: 100, debit: 30)

    expect(transaction.display).to eq '05/02/2019 ||  || 30.00 || 100.00 '
  end

  it 'should display date of transaction' do
    date = Time.new(2019, 2, 5)
    allow(Time).to receive(:now).and_return(date)
    transaction = Transaction.new(balance: 100, credit: 50)

    expect(transaction.display).to eq '05/02/2019 || 50.00 ||  || 100.00 '
  end
end
